### ISSUE TITLE
Possible error in Mystcraft fluid blacklist.

### ISSUE THREAD

**Issue by CombatZAK** [SlimeKnights/TinkersConstruct#1656]
I have spent the last few days investigating some anomalies in mod compatibility behavior between Tinkers' Construct and Mystcraft. It turns out the TConstruct implementation of blacklisting fluids via IMC exposed a hidden bug in the Mystcraft code that caused all fluids examined after a blacklisted fluid to also be removed, even if they weren't on the blacklist. For reference: https://github.com/XCompWiz/Mystcraft-Issues/issues/97

XCompWiz released the fix for this bug on 12-June. However, during that investigation, I believe I have uncovered two mistakes in the TinkerMystcraft class in Tinkers' Construct.

Here is the **init** method from TinkerMystcraft.java file:

```java
@Handler
public void init (FMLInitializationEvent event)
{
	final Fluid[] fluids = new Fluid[] {
			// precious tinker fluids
			moltenGoldFluid,
			moltenSteelFluid,
			moltenEmeraldFluid,
			moltenArditeFluid,
			moltenCobaltFluid,
			// all alloys
			pigIronFluid,
			moltenBronzeFluid,
			moltenAlumiteFluid,
			moltenAlubrassFluid,
			moltenManyullynFluid,
			// precious TE fluids
			moltenEnderFluid,
			moltenSilverFluid,
			moltenInvarFluid,
			moltenElectrumFluid,
			moltenShinyFluid,
			moltenSignalumFluid,
			moltenLumiumFluid,
			moltenMithrilFluid
	};

	TConstruct.logger.info("Mystcraft detected. Blacklisting Mystcraft fluid symbols.");
	for(Fluid fluid : fluids)
	{
		if(fluid == null)
			continue;
		FMLInterModComms.sendMessage("Mystcraft", "blacklistfluid", fluid.getName());
	}
}
```

The list includes one fluid that should probably not be there and is missing one fluid that probably should (very likely these things are related). The fluids in question are **moltenEnderFluid** and **moltenEnderiumFluid**, respectively.

**moltenEnderFluid** very likely *Resonant Ender* is a Thermal Foundation fluid, not registered for Tinker's Construct compatibility, and TConstruct probably doesn't have standing to blacklist it. That choice/responsibility should fall to Team CoFH.

By contrast, **moltenEnderiumFluid**, very likely *Molten Enderium* is an alloy fluid added for Thermal Foundation / Tinkers' Construct compatibility: a means for creating Enderium Alloy in a Tinker's Construct smeltery. I suspect this is the precious fluid that was intended for blacklisting in the TinkerMystcraft class.

I'll close this by saying that I have not had an opportunity to examine every facet of this issue. Any problems with blacklisting fluids in Mystcraft has been hidden for several months by the aforementioned Mystcraft bug. This included the unexpected blacklisting of materials that the Tinkers' Construct team did not intend to block - copper, tin, iron and aluminum, for example. However, I can confirm that even prior to the Mystcraft#97 fix, Molten Enderium was *not* blacklisted when TConstruct and Mystcraft were in the environment. In fact is was one of the two fluids that actually survived the Mystcraft anomaly - that is confirmation enough for me that even with the fix in place, oceans of Molten Enderium will probably still appear in Mystcraft ages.

TL;DR summary: TinkerMystcraft class should probably not include moltenEnderFluid and should definitely include moltenEnderiumFluid in its blacklist. Also IMC blacklisting should now work as intended on the Mystcraft side.

**Comment by bonii-xx** [comment]
TCon has its own molten ender. TFs resonant ender is the same. It being in there is intended.
Enderium is indeed missing. It was added very late.

**Comment by bonii-xx** [comment]
252662852e4bf7fec4a922e97f6b9acd8a92e0cc
